pkg/factory/machinepool.go

```diff
@@ -0,0 +1,105 @@
+package factory
+
+import (
+	"github.com/giantswarm/microerror"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	"github.com/giantswarm/conditions-handler/pkg/conditions/composite"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/creating"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/infrastructureready"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/summary"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/upgrading"
+	"github.com/giantswarm/conditions-handler/pkg/handler"
+)
+
+// NewMachinePoolConditionsHandler creates a composite handler for reconciling
+// MachinePool conditions, which consists of condition handlers for
+// InfrastructureReady, Ready, Creating and Upgrading conditions.
+func NewMachinePoolConditionsHandler(config handler.Config) (*composite.Handler, error) {
```

Comment:
Here we compose the conditions handler for the MachinePool controller, by combining multiple smaller condition handlers, for all conditions that we can/want to set on the MachinePool CR.